sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthorizationCodeCredential.java

```diff
@@ -63,7 +66,9 @@
                    cachedToken.set(new MsalAuthenticationAccount(
                                 new AuthenticationRecord(msalToken.getAuthenticationResult(),
                                         identityClient.getTenantId())));
-                   return  msalToken;
-               });
+                   return (AccessToken) msalToken;
+               })
+            .doOnNext(token -> LoggingUtil.logTokenSuccess(logger, request))
```

Comment:
do we want to replace this with doOnSuccess ?
and also add doOnRequest ?

Reply:
I modified everything to be doOnSuccess. We don't log anything on doOnRequest().